### ISSUE TITLE
Support for Non-Github projects?

### ISSUE THREAD

**Issue by jeeftor** [realm/jazzy#314]
Any chance we could have an "any-url" support here?  For say Atlassian Stash or BitBucket?  We run stash and gitlab internally at my office.

```
-g, --github_url URL             GitHub URL of this project (i.e. \
            https://github.com/realm/realm-cocoa)
    --github-file-prefix PREFIX  GitHub URL file prefix of this project (i.e. \
            https://github.com/realm/realm-cocoa/tree/v0.87.1)
```

**Comment by jpsim** [comment]
You could pass in `--github_url` and `--github-file-prefix` along with custom templates that can display whatever you want instead of "View on GitHub" and the little octicon glyph.

**Comment by jeeftor** [comment]
I created a pull request for this guy! :+1: 

https://github.com/realm/jazzy/pull/315

**Comment by istx25** [comment]
I'll try to get around to reviewing #315. Seems to still be open.

**Comment by arueda** [comment]
I see that PR #315 was closed. Is supporting projects outside github without using custom templates still a goal?

**Comment by jpsim** [comment]
> Is supporting projects outside github without using custom templates still a goal?

Yeah, that'd be great! I shared my thoughts a few years ago in that PR's thread:

> Taking a step back though, I think we can generalize things quite a bit by making the following changes:
> 1. `--github_url` -> `--project-url`
> 2. `--github-file-prefix` -> `--source-host-prefix`
> 3. `--source-host` with `github`, `bitbucket` & `gitlab` as options which would affect the ["View on GitHub"](https://github.com/realm/jazzy/blob/v0.3.1/lib/jazzy/templates/header.mustache#L4-L6) in the header and ["Show on GitHub"](https://github.com/realm/jazzy/blob/v0.3.1/lib/jazzy/templates/task.mustache#L60-L64) link in the declaration description

**Comment by johnfairh** [comment]
Addressed in master via #1268
